### DISCUSSION TITLE
A tip for compling and installing mitsuba3 locally from git

### DISCUSSION THREAD

**Discussion by lynshwoo2022** [mitsuba-renderer/mitsuba3#473] [General]
in case anyone new to github like me want to use the latest mitsuba3 which haven't be released on pypi yet. here's my experience on how to compile it from git :)
1. ```git clone xxx.git --recursive``` or download everything manually including `ext` files.
2. install packages required in `pyproject.toml`
3. `cd /ext/drjit/`->`python setup.py build`->`python setup.py install`
4. `cd /ext/zlib/` -> `chmod 777 -R configure` -> `./configure` ->`make test` ->`make install`
5. back to `mitsuba3-mater/`->`python setup.py build` ->`python setup.py install`

**Comment by njroussel** [comment]
Note that step 4) is only necessary on Windows.

**Reply by njroussel** [reply]
Let me clarify:

This local install process is not actively supported. At any point between version releases, the Mitsuba3 might be using a DrJit version that is not compatible with the latest PyPI release. Unfortunately, `pip install ./mitsuba3` or `python setup.py install` will both try to use the latest DrJit release rather than compiling whatever commit the Mitsuba3 repository is pointing to.
Following the compilation [instructions in the documentation](https://mitsuba.readthedocs.io/en/stable/src/developer_guide/compiling.html#linux) is the preferred approach. Modifying the `PATH/PYTHONPATH` globally for your system should functionally be the same as if you had used `pip install` on the project.

**Comment by lynshwoo2022** [comment]
one more hint, if we want to install it under a specific conda virtual environment, we need to specify the directory for mitsuba like:
```
python setup.py install --prefix=/anaconda/envs/virtual_env_name/
```

**Comment by pekCs** [comment]
When I used `python setup.py build`,I failed to build it.As it follows:

 FAILED: Release/python/mitsuba/__init__.pyi D:/mitsuba3/mitsuba3/_skbuild/win-amd64-3.8/cmake-build/Release/python/mitsuba/__init__.pyi
cmd.exe /C "cd /D D:\mitsuba3\mitsuba3\_skbuild\win-amd64-3.8\cmake-build\src\python && "C:\Program Files\CMake\bin\cmake.exe" -E env PYTHONPATH=D:\mitsuba3\mitsuba3\_skbuild\win-amd64-3.8\cmake-build\Release\python; D:/miniconda/python.exe -Xutf8 D:/mitsuba3/mitsuba3/src/python/../../resources/generate_stub_files.py D:/mitsuba3/mitsuba3/_skbuild/win-amd64-3.8/cmake-build/Release/python/mitsuba"
Exit code 0xc0000409

ninja: build stopped: subcommand failed.
Traceback (most recent call last):
  File "D:\miniconda\lib\site-packages\skbuild\setuptools_wrap.py", line 674, in setup
    cmkr.make(make_args, install_target=cmake_install_target, env=env)
  File "D:\miniconda\lib\site-packages\skbuild\cmaker.py", line 697, in make
    self.make_impl(clargs=clargs, config=config, source_dir=source_dir, install_target=install_target, env=env)
  File "D:\miniconda\lib\site-packages\skbuild\cmaker.py", line 742, in make_impl
    raise SKBuildError(msg)

An error occurred while building with CMake.
  Command:
    'C:\Program Files\CMake\bin\cmake.EXE' --build . --target install --config Release --
  Install target:
    install
  Source directory:
    D:\mitsuba3\mitsuba3
  Working directory:
    D:\mitsuba3\mitsuba3\_skbuild\win-amd64-3.8\cmake-build
Please check the install target is valid and see CMake's output for more information.

How can I fix it?

**Reply by njroussel** [reply]
See the amended answer I provided above: https://github.com/mitsuba-renderer/mitsuba3/discussions/473#discussioncomment-4555866
